src/DataTypes/DataTypeEnum.cpp

```diff
@@ -78,7 +79,19 @@ Field DataTypeEnum<Type>::getDefault() const
 template <typename Type>
 void DataTypeEnum<Type>::insertDefaultInto(IColumn & column) const
 {
-    assert_cast<ColumnType &>(column).getData().push_back(this->getValues().front().second);
+    const auto & default_value = this->getValues().front().second;
+
+    if (auto * sparse_column = typeid_cast<ColumnSparse *>(&column))
```

Comment:
This code is actually bad, but unfortunately, `IDataType::insertDefaultInto` breaks the abstraction of the separation of data types, serializations, and columns. Since this method is overridden only for `DataTypeEnum` and this code has remained unchanged for years, it should be okay.